logstreams/src/main/java/io/zeebe/logstreams/impl/delete/LogStreamDeletionService.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright © 2017 camunda services GmbH (info@camunda.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.zeebe.logstreams.impl.delete;
+
+import io.zeebe.logstreams.impl.Loggers;
+import io.zeebe.logstreams.impl.log.index.LogBlockIndex;
+import io.zeebe.logstreams.impl.log.index.LogBlockIndexContext;
+import io.zeebe.logstreams.log.LogStream;
+import io.zeebe.logstreams.spi.LogStorage;
+
+public abstract class LogStreamDeletionService implements DeletionService {
+  protected LogStream logStream;
+  protected final LogStorage logStorage;
+  protected final LogBlockIndex logBlockIndex;
+  protected final LogBlockIndexContext blockIndexContext;
+
+  public LogStreamDeletionService(LogStream logStream) {
+    logBlockIndex = logStream.getLogBlockIndex();
+    logStorage = logStream.getLogStorage();
+    blockIndexContext = logBlockIndex.createLogBlockIndexContext();
+  }
+
+  public void delete(long position) {
+    position = Math.min(position, getMinimumExportedPosition());
+
```

Comment:
All lines below should be moved backed to the log stream. The log stream is responsible for the log block index and log storage, we should not modify it outside. It is more an implementation detail of how the log stream handles the deletion.

Then we probably also have no need for this inheritance, if only consist of `
    position = Math.min(position, getMinimumExportedPosition());`

Reply:
I think the issue is more on the dependencies. If the logstream does the deletion, then it depends on the exporter service which then depends on the partition service. It would work if we leave the exporter supplier but that was one of the main hacks we wanted to remove with this issue

Reply:
Hm ? The logstream only gets a upper bound position to which it should delete. No need to know where this came from

Reply:
Ah you mean moving only the index and storage delete, not the whole thing, my bad. Yes, that makes sense. About the inheritance, it's only relevant because the way we get the minimum exported position changes from follower to leader, I think that doesn't change right?

Reply:
Ok :+1: it does not change, but I think we do not need to have inheritance here just use the logstream in both classes and implement the interface. You have the leader deletion service, which implements the interface and resolves the position and calls log stream delete. And we have the FollowerDeletionService which implements the interface, resolves the position and calls the log stream delete.